### ISSUE TITLE
Lack of keyboard support

### ISSUE THREAD

**Issue by jducrot** [cortex-js/cortexjs.io#27]
Although the documentation implies this library offers keyboard support with and without the use of a screen reader, I can't find any markup that supports that claim. Even the demo on your site is not keyboard accessible. I don't see any options to configure that on my end. Is there a specific version of the library that we need to loand to get a keyboard accessible keyboard?

**Comment by arnog** [comment]
The virtual keyboard itself is not keyboard navigable, but all the functionality of the virtual keyboard can be accessed using a physical keyboard. For example, arbitrary LaTeX markup can be entered using the LaTeX mode, which can be accessed by pressing the ESC or `/` keys.

**Comment by jducrot** [comment]
How do you communicate that information to assistive technology users? Leaving Latex aside, if I want to write a fraction as a blind user, how would I go about triggering that functionality using only my keyboard? Finally, why not make the virtual keyboard keyboard accessible? You already have some of the markup in place since you are using HTML. If the problem is guidance and testing I could help with that because we need more accessible virtual keyboard solutions available to developers.

**Comment by arnog** [comment]
> How do you communicate that information to assistive technology users? 

I had some voice feedback at some point that explained this, but the feedback I got was that it was too chatty and not necessary.

> Leaving Latex aside, if I want to write a fraction as a blind user, how would I go about triggering that functionality using only my keyboard?

By pressing the "/" key, i.e. "1/2".

There are a number of non-LaTeX related physical keyboard shortcuts available: https://cortexjs.io/mathlive/reference/keybindings/

> Finally, why not make the virtual keyboard keyboard accessible?

When I tested this with people using screen readers to navigate, the feedback I got was that making the virtual keyboard navigable was pointless. They would much rather type "/" to insert a fraction than activate the virtual keyboard and tap "tab tab tab down-arrow down-arrow return" to do the same thing.

**Comment by jducrot** [comment]
While the feedback you received is definitely useful perspective, ss a certified accessiblity professional with 20 years in software development I can tell you that either the keyboard shortcuts should be referenceable from the keyboard (contextual help is actually a new criteria in WCAG 2.2 which is coming out in the next few months) or the virtual keyboard must be accessible, ideally both. While some users familiar with the app might prefer typing "/" to create a fraction, some less experience users or users of some assistive technologies like switches might not have that ability, hence the need to support both approaches. I am provividing this feedback with the genuine intention to make you great library better and I would be happy to provide guidance and support. You've already done a tremendous amount of work, and this work would put this library over the top. Unfortunately saying that this keyboard is accessible to assistive technologies in its current state is midleading some devs into using it thinking there is nothing else to account for.

**Comment by arnog** [comment]
@jducrot thanks for the feedback. Could you let me know where it is indicated that the virtual keyboard is accessible to assistive technologies, because I agree that is not the case and I don't want to give a false impression. The mathfield itself is accessible to screen readers and uses ARIA live regions and its own built-in math to speech engine to help with navigation of the input, but the virtual keyboard is an alternate optional input method (on the desktop) and doesn't have any special accessibility support.

Could you point me to the WCAG 2.2 criteria on contextual help? That sounds interesting, but I can only find 3.2.6 on consistent help.

Given my limited time, I've prioritized improving the accessibility of the mathfield based on user feedback rather than going for a checkmark compliance with section 508 or other standards. The mathfield has successfully been used by users using assistive technology, so that's a win for me. That said, I will gladly consider submissions to the projects in the form of Pull Requests for anyone who wants to improve further the accessibility of the component.

**Comment by jducrot** [comment]
Here is the link to [WCAG 2.2 SC 3.2.6 A consistent help](https://www.w3.org/WAI/WCAG22/Understanding/consistent-help.html) and a link to[ WCAG 2.2 SC 3.3.5 AAA Help (context sensitive)](https://www.w3.org/WAI/WCAG22/Understanding/help.html)

**Comment by arnog** [comment]
Great. I've added a section with the (major) keyboard shortcuts and a link to that section right next to the mathfield: https://cortexjs.io/mathlive/demo/
